airbyte-cdk/python/airbyte_cdk/sources/declarative/retrievers/simple_retriever.py

```diff
@@ -62,6 +64,7 @@ class SimpleRetriever(Retriever, HttpStream):
     _primary_key: str = field(init=False, repr=False, default="")
     paginator: Optional[Paginator] = None
     stream_slicer: Optional[StreamSlicer] = SinglePartitionRouter(parameters={})
```

Comment:
would it make sense to change the type of this field to `PartitionRouter` and completely decouple the cursor from the partition routers in the retriever?

Reply:
Not exactly as if there is a cursor, it'll be the stream slicer (see https://github.com/airbytehq/airbyte/pull/27398/files#diff-bbfbb380b441567d8e40967c149ae0397e41357a213f05add6d3b72cf92a91fdR794-R795). As of today, I think the simple retriever doesn't know about partitions. Does that sound weird to you?